src/app/components/UserSettingsModal.tsx

```diff
@@ -0,0 +1,153 @@
+import {
+  Modal,
+  ModalOverlay,
+  ModalContent,
+  ModalCloseButton,
+  ModalBody,
+  ModalProps,
+  Heading,
+  useToast,
+} from "@chakra-ui/react";
+import { useEffect, useState } from "react";
+import { userDataByAddress } from "../store/atoms/userDataByAddress";
+import { useRecoilState } from "recoil";
+import { gasHawkClient } from "../interface/gashawk-backend";
+import { ModalSpinner } from "./ModalSpinner";
+import { SettingsAutoQueue } from "./SettingsAutoQueue";
+import {
+  SettingsDefaultDeadline,
+  stepValues,
+  displayValues,
+} from "./SettingsDefaultDeadline";
+import { SettingsFlashbots } from "./SettingsFlashbots";
+import { SettingsPreventTxStuck } from "./SettingsPreventTxStuck";
+import { SettingsPrioritizeLowFees } from "./SettingsPrioritizeLowFees";
+import { SettingsProbeTx } from "./SettingsProbeTx";
+import { defaultRpcType } from "./SettingsFlashbots";
+import { SettingsSaveChanges, toastStatusType } from "./SettingsSaveChanges";
+
+export const UserSettingsModal = (props: Omit<ModalProps, "children">) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [sliderValue, setSliderValue] = useState(300000);
+  const [sliderDefaultValue, setSliderDefaultValue] = useState(0);
+  const [{ address: user }] = useRecoilState(userDataByAddress);
+  const [autoQueue, setAutoQueue] = useState(false);
+  const [flashbots, setFlashbots] = useState<defaultRpcType>("");
+  const [
+    holdBackIfBaseFeeExceedsBfCreatedAt,
+    setHoldBackIfBaseFeeExceedsBfCreatedAt,
+  ] = useState(false);
+  const [preventStuckInMempool, setPreventStuckInMempool] = useState(false);
+  const [probeTxBeforeSubmission, setProbeTxBeforeSubmission] = useState(false);
+  const [deadlineDisplay, setDeadlineDisplay] = useState<string>(
+    displayValues[0]
+  );
+  const toast = useToast();
+  const showToast = (message: string, status: toastStatusType) => {
+    toast({
+      title: "Settings",
+      description: message,
+      status: status, // or "error", "warning", "info", "success"
+      duration: 3000, // milliseconds
+      isClosable: true,
+    });
```

Comment:
Please avoid useless comments:
```suggestion
    toast({
      title: "Settings",
      description: message,
      status: status,
      duration: 3000,
      isClosable: true,
    });
```

Reply:
Removed comments. [3561f36](https://github.com/sohjag/gashawk-app-ext/pull/5/commits/3561f3605f030befde37958c9cb809c8c5db274b)